swarm_dial.go

```diff
@@ -352,6 +349,157 @@ func (s *Swarm) filterKnownUndialables(addrs []ma.Multiaddr) []ma.Multiaddr {
 	)
 }
 
+func (s *Swarm) scheduleDialAddrs(ctx context.Context, addrs []ma.Multiaddr) <-chan ma.Multiaddr {
+	var immediateAddrs []ma.Multiaddr
+	relayAddrs := make(map[peer.ID][]ma.Multiaddr)
+
+	for _, a := range addrs {
+		_, err := a.ValueForProtocol(P_CIRCUIT)
+		if err != nil {
+			immediateAddrs = append(immediateAddrs, a)
+			continue
+		}
+
+		relayS, err := a.ValueForProtocol(ma.P_P2P)
+		if err != nil {
+			// uspecfic relay addr; push it to immediate -- note that these addrs should not be
+			// advertised any more, but we don't want to break older nodes or tests
+			immediateAddrs = append(immediateAddrs, a)
```

Comment:
Shouldn't we just drop this?

Reply:
Maybe, I really don't know what to do with these!

Reply:
Maybe we should schedule them for last, after all specific relay addrs have been scheduled.

Reply:
Do we really gain anything? This literally means: dial me through random relays.

Reply:
Probably not. 
Only concern is that we do "support" users requesting dials with `ipfs swarm connect /p2p-circuit/ipfs/QmTarget`, and we may not want to completely break that.

Reply:
Gah. What if we just hack that into a higher layer? Or just drop it? Nobody should be relying on that in production and I don't feel all that bad about breaking demos.

Reply:
Maybe we should drop it -- but we might want to disable circuit.OptDiscovery in ipfs as well.
We can substitute it with `/ipfs/QmRelay/p2p-circuit/ipfs/QmTarget` anyway.